Date: Wed, 30 Mar 2011 14:11:09 +0530
From: Kamalesh Babulal <kamalesh@linux.vnet.ibm.com>
Subject: powerpc: Fix default_machine_crash_shutdown #ifdef botch

commit c2be05481f6125254c45b78f334d4dd09c701c82 upstream.

Introducing #ifdef to fix the build failure caused by
crash_kexec_wait_realmode(), with powerpc build with !SMP.

Reported-by: Ben Hutchings <ben@decadent.org.uk>
Signed-off-by: Kamalesh Babulal <kamalesh@linux.vnet.ibm.com>
cc: Paul E. McKenney <paulmck@linux.vnet.ibm.com>
cc: Michael Neuling <mikey@neuling.org>
cc: Benjamin Herrenschmidt <benh@kernel.crashing.org>
cc: Anton Blanchard <anton@samba.org>
---
 arch/powerpc/kernel/crash.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/arch/powerpc/kernel/crash.c b/arch/powerpc/kernel/crash.c
index fe02e71..6ff7701 100644
--- a/arch/powerpc/kernel/crash.c
+++ b/arch/powerpc/kernel/crash.c
@@ -445,7 +445,9 @@ void default_machine_crash_shutdown(struct pt_regs *regs)
	crash_kexec_prepare_cpus(crashing_cpu);
	cpu_set(crashing_cpu, cpus_in_crash);
	crash_kexec_stop_spus();
+#ifdef CONFIG_SMP
	crash_kexec_wait_realmode(crashing_cpu);
+#endif
	if (ppc_md.kexec_cpu_down)
		ppc_md.kexec_cpu_down(1, 0);
 }
